Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -77,21 +82,25 @@ section Zero
 
 variable [Zero R]
 
-instance : CoeFun (ArithmeticFunction R) fun _ => ℕ → R :=
-  ZeroHom.hasCoeToFun
+instance : FunLike (ArithmeticFunction R) ℕ fun _ ↦ R :=
+  inferInstanceAs (FunLike (ZeroHom ℕ R) ℕ fun _ ↦ R)
 
 @[simp]
-theorem toFun_eq (f : ArithmeticFunction R) : f.toFun = f :=
+theorem toFun_eq (f : ArithmeticFunction R) : f.toFun = f := by
```

Comment:
Why do you need `by` here?

Reply:
Removed `by` and moved `rfl` on previous line.